search_engine_parser/core/utils.py

```diff
@@ -33,10 +33,11 @@ def get_rand_user_agent():
 
 class CacheHandler:
     def __init__(self):
-        if not os.path.exists(os.path.join(FILEPATH, "cache")):
-            os.makedirs("cache")
         self.cache = os.path.join(FILEPATH, "cache")
-        enginelist = os.listdir(os.path.join(FILEPATH, "engines"))
+        engine_path = os.path.join(FILEPATH, "engines")
+        if not os.path.exists(self.cache):
```

Comment:
how is this implementation different from the former?

Reply:
The former one was os.makedirs("cache") not os.makedirs(self.cache)